core/trino-main/src/main/java/io/trino/operator/OrderByOperator.java

```diff
@@ -205,6 +205,9 @@ public void finish()
             return;
         }
         checkSuccess(spillInProgress, "spilling failed");
+        if (finishMemoryRevoke.isPresent()) {
```

Comment:
nit: Add a comment why `finishMemoryRevoke` might be present while `spillInProgress` is already done

Reply:
anyone can find the answer quickly

Reply:
Agreed. OTOH novice readers shouldn't read spilling operators' code. There be dragons.

Reply:
I'm pretty sure in 6 months you will loose context too ;)

Reply:
I just copied the logic from https://github.com/trinodb/trino/blob/701c0fa1b7bc2ac95e5c8db33219e032a36bb364/core/trino-main/src/main/java/io/trino/operator/join/HashBuilderOperator.java#L447-L450
if you think a comment is due, you can add in both places. 

- I am not doing this, because I think on the same reasoning we would need to add pretty lot more comments (around spillInProgress futures, etc.)